src/wwwroot/app/services/ReportsService.js

```diff
@@ -26,6 +26,12 @@ export default class ReportsService {
          .then(this._convertFromServerDates);
    }
 
+   getDataForCandidatesReport(searchParameters) {
+      searchParameters = this._convertToServerDates(searchParameters);
+      let candidatesReportUrl = 'candidateProgressReport';
+      return _HttpService.get(`${REPORT_URL}/${candidatesReportUrl}`, searchParameters);
+   }
+
```

Comment:
please use something like _ReportsService = this, then _ReportsService._convertToServerDates. And move move _convertToServerDates from service API